### PR TITLE
Make sure chosen interface has address

### LINKED ISSUE #3336: Chosen interface has no ip address caused kube-proxy failed to start
kube-proxy finds the first up interface (non p2p, loopback) as its host interface, but that interface doesn't have an ip address on my machine.

https://github.com/GoogleCloudPlatform/kubernetes/blob/53a9d106c4aabcd550cc32ae4e8004f32fb0ae7b/pkg/proxy/proxier.go#L760

Specifically, kube-proxy finds eth0, which doesn't have ip address associated with it (wlan0 does).  I don't know why eth0 no longer gets an ip address from dhcp server, it worked before.

### PR DESCRIPTION
**Pull request by deyuanix**
Fixes #3336

### CODE REVIEW

**Comment by thockin** [comment]
Was this a real problem?  What interface was up but has no address?

**Comment by thockin** [comment]
Tests failed.

--- FAIL: TestTCPProxy (0.00s)
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

**Comment by ddysher** [comment]
Yeah, it's a problem on my local machine, eth0 is up but has no address.  See #3336 

The 'index out of range' error is exactly what I'm trying to fix. Tests succeeded locally (failed without the fix).  I'll try to investigate more.

File: pkg/proxy/proxier.go [hunk 1]
```diff
@@ -764,17 +764,24 @@ func chooseHostInterface() (net.IP, error) {
...
+				break
+			}
 		}
 	}
 	if i == len(intfs) {
 		return nil, err
 	}
 	glog.V(2).Infof("Choosing interface %s for from-host portals", intfs[i].Name)
-	addrs, err := intfs[i].Addrs()
+	addrs, err = intfs[i].Addrs()
```
Comment (ddysher) [inline pkg/proxy/proxier.go]: Have to get the address again... I think theoretically, addres should already have the value.

**Comment by ddysher** [comment]
friendly ping!  This keeps failing on my machine...

**Pull request closed without merging**